the-collab-lab/tcl-37-smart-shopping-list
# Mark item as purchased

## Description

In this PR, we added the functionality to mark an item as purchased for 24 hours. To do this, we utilized the `Moment.js` library to capture the current time and calculate time differences in hours.

When an item is clicked, we used an `updateDocument()` function to update the `purchased_date` field for that specific document in the database. We then created a helper function `calcTimeDiff()` to compare the `purchased_date` time to the current time. If the calculated time, is less than 24 hours, the item checkbox will remain checked and disabled. 




## Related Issue
Closes #8 



## Acceptance Criteria
AC:

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Video Demo


https://user-images.githubusercontent.com/67841237/152593482-82622930-9f3a-4263-84e4-dc925f16022c.mp4









## Testing Steps / QA Criteria

1. npm install (we imported the Moment.js package)
2. npm start
3. Navigate to a list via a three word token. Suggested: "alba terre tussle" or "vary adore suite"
4. Notice that some checked items are disabled.  This means they have been purchased within the last 24 hours.
5. Check an item from the list.
6. Go into the Firestore database and confirm the purchased_date field is updated.
7. You can edit the purchased_date field to test if the item unchecks after 24 hours.
